pkg/apis/core/v1beta1/types.go

```diff
@@ -362,6 +382,123 @@ type StorageProfileList struct {
 	Items []StorageProfile `json:"items"`
 }
 
+// DataSource references an import/clone source for a DataVolume
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+type DataSource struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   DataSourceSpec   `json:"spec"`
+	Status DataSourceStatus `json:"status,omitempty"`
+}
+
+// DataSourceSpec defines specification for DataSource
+type DataSourceSpec struct {
+	// Source is the source of the data referenced by the DataSource
+	Source DataSourceSource `json:"source"`
+}
+
+// DataSourceSource represents the source for our DataSource
+type DataSourceSource struct {
+	PVC *DataSourceSourcePVC `json:"pvc"`
```

Comment:
omitempty as this will eventually be optional

Reply:
How can it be optional when it is currently the only possible source which is a must?
When we'll have other optional sources here all of them will sure be `optional`.

Reply:
IMO all pointer fields should be optional.  Eventually the fields of this struct will have to be validated in a webhook to enforce a "one of" constraint.  Whether you do that now or later is your choice. If you don't do that now, there should at least be a comment here.

Reply:
Will fix

Reply:
fixed